phper/src/builder/data_type.rs

```diff
@@ -0,0 +1,43 @@
+pub enum DataType {
+    // PHP < 8.2: mixed
+    // PHP >= 8.2: null
+    Null,
+    // T1|T2|T3...
+    Union(Vec<DataType>),
+    // float
+    Float,
+    // int
+    Integer,
+    // int
+    PostivieInteger,
+    // int
+    RangedInteger(i64, i64),
+    // string
+    String,
+    // string
+    NonEmptyString,
+    // string
+    Classname,
+    // bool
+    Boolean,
+    // true
+    True,
+    // false
+    False,
+    // array
+    HashTable,
+    // array
+    List(Vec<DataType>),
+    // array
+    Array(Vec<DataType>),
+    // T
+    Object(String),
+    // T
+    Enum(String),
```

Comment:
this is the same as `Object(String)`, with only 1 difference, when validating `ArgumentType`, the `default` value of can either be `None` or `Some(ConstantValue::EnumCase(A, B))`, where `A` is the same as the string provided for the name of the enum.